src/NexxLogic.BlobConfiguration.AspNetCore/FileProvider/BlobFileProvider.cs

```diff
@@ -84,11 +129,51 @@ public IDirectoryContents GetDirectoryContents(string subpath)
 
     public IChangeToken Watch(string filter)
     {
+        if (_disposed) throw new ObjectDisposedException(nameof(BlobFileProvider));
+
+        // Use enhanced change token if available, otherwise fall back to legacy implementation
+        if (_blobServiceClient != null && _useContentBasedDetection)
+        {
+            if (_enableDetailedLogging)
+            {
+                _logger.LogDebug("Creating enhanced watch token for filter: {Filter} with debounce: {Debounce}s", 
+                    filter, _debounceDelay.TotalSeconds);
+            }
```

Comment:
Do we really need a custom, non-standard flag to control detailed logging? I think we should be able to just use log level configuration from appsettings.json.